tests/lib/rules-preprocessor/gjs-gts-parser-test.js

```diff
@@ -761,4 +761,80 @@ describe('multiple tokens in same file', () => {
     expect(resultErrors[2].message).toBe("'bar' is not defined.");
     expect(resultErrors[2].line).toBe(17);
   });
+
+  it('lints while being type aware', async () => {
+    const eslint = new ESLint({
+      ignore: false,
+      useEslintrc: false,
+      plugins: { ember: plugin },
+      overrideConfig: {
+        root: true,
+        env: {
+          browser: true,
+        },
+        plugins: ['ember'],
+        extends: ['plugin:ember/recommended'],
+        overrides: [
+          {
+            files: ['**/*.gts'],
+            parser: 'eslint-plugin-ember/gjs-gts-parser',
+            parserOptions: {
+              project: './tsconfig.eslint.json',
+              tsconfigRootDir: __dirname,
+              extraFileExtensions: ['.gts'],
+            },
+            extends: [
+              'plugin:@typescript-eslint/recommended-requiring-type-checking',
+              'plugin:ember/recommended',
+            ],
+            rules: {
+              'no-trailing-spaces': 'error',
+              '@typescript-eslint/prefer-string-starts-ends-with': 'error',
+            },
+          },
+          {
+            files: ['**/*.ts'],
+            parser: '@typescript-eslint/parser',
```

Comment:
@patricklx don't we need to leave `*.ts` files parsing to `eslint-plugin-ember/gjs-gts-parser` also?

Reply:
If we don't use it then I assume we will again have errors if we try to import from a `.gts` file inside a regular `.ts` file

Reply:
Though this means that we will need to recommend ppl use `eslint-plugin-ember/gjs-gts-parser` for all files `js/ts/gjs/gts` in order to have it work properly

Reply:
aha I see, so basically we just monkey patch `ts` directly. Might be a bit brittle but it will work :D.
And we don't need to worry about creating our own TS program nor involving glint :D
Solid idea!

Reply:
the problem i'm facing now is how to replace `.gts` without breaking some eslint rule...
especially the stylistic ones.
it would shift all code by 1 character... but adding something after it breaks some whitespace rules.
So now i use the extension `.mts`

Reply:
> Though this means that we will need to recommend ppl use `eslint-plugin-ember/gjs-gts-parser` for all files `js/ts/gjs/gts` in order to have it work properly

Maybe, now ts.sys is overwritten as soon as gjs-gts-parser is loaded. Which is loaded during config loading.

Reply:
@patricklx yes you are right, with ts patch approach we can leave `ts` files to `typescript-eslint/parser`.
Again solid idea patching this :)